src/pain/001/SEPACreditPaymentInitiation.ts

```diff
@@ -0,0 +1,148 @@
+    return {
+      ...instruction,
+      dinero: Dinero({ amount: instruction.amount, currency: instruction.currency }),
+    };
+  }
+
+  // XXX: Does not work with different currencies
+  private sumPaymentInstructions(instructions: AtLeastOne<SEPACreditPaymentInstruction>): string {
+    return instructions.reduce(
+      (acc: Dinero.Dinero, next): Dinero.Dinero => {
+        return acc.add(next.dinero as Dinero.Dinero);
+      },
+      Dinero({ amount: 0, currency: instructions[0].currency }),
+    ).toFormat('0.00');
+  }
+
+  private validate() {
+    if (this.messageId.length > 35) {
+      throw new Error('messageId must not exceed 35 characters');
+    }
+
+    const creditorWithoutAddress = this.paymentInstructions.find(
+      instruction => !instruction.creditor.address,
+    );
+
+    const creditorWithIncompleteAddress = this.paymentInstructions.find(
+      instruction => {
+        const address = instruction.creditor.address;
+        return !address || !address.country;
+      },
+    );
+
+    if (creditorWithIncompleteAddress) {
+      throw new Error(
+        'All creditors must have complete addresses (street name, building number, postal code, town name, and country)',
+      );
+    }
+  }
+
+  paymentInformation(instruction: SEPACreditPaymentInstruction) {
+    const paymentInfoID = sanitize(instruction.id || uuidv4(), 35);
+    instruction.dinero ??= Dinero({ amount: instruction.amount, currency: instruction.currency });
+
+    return {
+      PmtId: {
+        InstrId: paymentInfoID,
+        EndToEndId: paymentInfoID,
+      },
+      Amt: {
+        InstdAmt: {
+          '#': instruction.dinero.toFormat('0.00'),
+          '@Ccy': instruction.currency,
+        },
+      },
+      CdtrAgt: this.agent(instruction.creditor.agent as BICAgent),
+      Cdtr: this.party(instruction.creditor as Party),
+      CdtrAcct: {
+        Id: { IBAN: (instruction.creditor.account as IBANAccount).iban },
+        Ccy: instruction.currency,
+      },
+      RmtInf: instruction.remittanceInformation ? {
+        Ustrd: instruction.remittanceInformation,
+      } : undefined,
+    };
+  }
+
+  public serialize(): string {
+    const xml = {
+      Document: {
+        '@xmlns': 'urn:iso:std:iso:20022:tech:xsd:pain.001.001.03',
+        '@xmlns:xsi': 'http://www.w3.org/2001/XMLSchema-instance',
+        CstmrCdtTrfInitn: {
+          GrpHdr: {
+            MsgId: this.messageId,
+            CreDtTm: this.creationDate.toISOString(),
+            NbOfTxs: this.paymentInstructions.length.toString(),
+            CtrlSum: this.paymentSum,
+            InitgPty: {
+              Nm: this.initiatingParty.name,
+              Id: {
+                OrgId: {
+                  Othr: {
+                    Id: this.initiatingParty.id,
+                  },
+                },
+              },
+            },
+          },
+          PmtInf: {
+            PmtInfId: `TRF${Date.now()}`,
```

Comment:
Can we use UUID as a default here? This is what we did for SWIFT I believe - may be a more durable solution

Reply:
I think this is good. It will probably throw issues ONLY if we create multiple payments wihtin the same millisecond

Reply:
I was worried about that. It's hacky. I was in the process of changing it to a UUID. We could still do that.

Reply:
It's okay as it is, but I believe we can make it even better if we use the UUID. 

I think the long term term. There should be an optional `paymentInformationId?: string` that should exist within the InitiationConfig, for example. 

For our usecase, I think we do not need this yet, default to Epoch is good, UUID is perfect.